dart/lib/src/observers.dart

```diff
@@ -0,0 +1,12 @@
+import 'dart:async';
+
+import '../sentry.dart';
+
+// This file will contain observer definitions that are executed during
+// specific points in the SDK such as as right before an event is sent.
+
+/// Called right before an event is sent, after all processing is complete.
+/// Should not modify the event at this point.
+abstract class BeforeSendEventObserver {
```

Comment:
unsure how to call this, JS calls them hooks

Reply:
it's the same concept as JS hook that is only used internally to do additional code execution